packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -361,6 +379,12 @@ export class TaskDefinition extends TaskDefinitionBase {
    * Adds a new container to the task definition.
    */
   public addContainer(id: string, props: ContainerDefinitionOptions) {
+    if (this.cpu !== undefined && props.cpu !== undefined && parseInt(this.cpu, 10) < props.cpu) {
```

Comment:
You also have to take into account that the CPU and Memory values could be Tokens. If that's the case, you can't do any validation at all.

Reply:
Would we want to do summing of all added containers as well?

Reply:
If you know for a fact that doing this will cause the stack to be undeployable, might as well make it an error.

Reply:
> Would we want to do summing of all added containers as well?

Not sure how we want to handle that. These are maximum limits, so in theory you could have the max limits of a container exceed the task's limits, but still be able to run successfully.

Reply:
> If you know for a fact that doing this will cause the stack to be undeployable, might as well make it an error.

This is the current behaviour, should we just handle for the current behaviour and update it in the future if there are changes that make this obsolete? (understanding that there will be a delay between the service making a change and the cdk supporting it.)

Reply:
> You also have to take into account that the CPU and Memory values could be Tokens. If that's the case, you can't do any validation at all.

Does it make sense to add validation in the cases it's not a token, and throw an error, but have no validation if it's tokenized?

Reply:
Yes.

Reply:
> This is the current behaviour, should we just handle for the current behaviour and update it in the future if there are changes that make this obsolete?

Do you mean that in the future, this may not be a hard error? If that is true, then you're right, a warning would be better.





> Does it make sense to add validation in the cases it's not a token, and throw an error, but have no validation if it's tokenized?

Exactly what I was thinking!

Reply:
> > Does it make sense to add validation in the cases it's not a token, and throw an error, but have no validation if it's tokenized?
> 
> Exactly what I was thinking!

I'll close this PR and the associated issue since it doesn't make sense to have two behaviours.